### PR TITLE
[15.0] [MIG] l10n_fi_bank_barcode, l10n_fi_invoice

### PR DESCRIPTION
**Pull request by nedaszilinskas**

### CODE REVIEW

File: l10n_fi_invoice/models/account_move.py [hunk 1]
```diff
@@ -18,21 +18,12 @@ class AccountMove(models.Model):
...
         filename = '-'.join((
             type_string,
             invoice_numbers,
             self.company_id.display_name,
-            self.partner_id.display_name)). \
+            self.partner_id.display_name or '')). \
             replace(' ', '-').replace(',', '').replace('--', '-')
         return filename
-
-    def invoice_print(self):
```
Comment (nedaszilinskas) [inline l10n_fi_invoice/models/account_move.py:31]: I think this is never called in 13.0. This was probably supposed to be `action_invoice_print` ([link to code](https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_move.py#L2514)). I have removed it here as it doesn't seem to be used anyway. What do you think about it?

Comment (aisopuro) [reply l10n_fi_invoice/models/account_move.py:31]: Yeah, probably a leftover from older Odoo versions: in 10 it was still called just `invoice_print` https://github.com/odoo/odoo/blob/10.0/addons/account/models/account_invoice.py#L394

Seems fine to me. If you're migrating this for some specific project, you might want to ensure that the expected user flows still work. This seems to have been intended to make it so that when you press "Print & Send", it would use the finnish invoice PDF by default. It appears to have been broken in 13, but now we're removing it explicitly. So you might want to make sure the user flow is still as expected: it isn't too difficult to get this PDF printed when sending invoices?

**Comment by aisopuro** [comment]
We recently got the issue #22 which indicated that `l10n_fi_invoice` could perhaps do a better job of indicating how it should be configured/used in order to get the invoice print in different contexts (for example to be the default attachment when sending invoices by email). Do you think it would be possible to add a section into the README that says a bit more about how to use the module?

**Review by aisopuro — CHANGES_REQUESTED** [review]
If possible, I'd like to see the changes split into two commits, one for each module. Should be as simple as
```
git reset origin/15.0
git add l10n_fi_bank_barcode
git commit ...
git add l10n_fi_invoice
git commit ...
```

File: l10n_fi_invoice/models/account_move.py (hunk 1 above)
Comment (nedaszilinskas) [reply l10n_fi_invoice/models/account_move.py:31]: @aisopuro I've added your comment/explanation from that issue to a readme.

As for myself, it was a little bit of a struggle at first to get a barcode to print (as it depends on so many conditions, needed to look into code directly to follow). Do you think we should also list these conditions in `l10n_fi_invoice` readme (barcode func. comes from another module - `l10n_fi_bank_barcode`)?

It is added here though https://github.com/avoinsystems/avoinsystems-addons/blob/13.0/l10n_fi_bank_barcode/__manifest__.py#L28. Do you think that's enough?

**Review by aisopuro — APPROVED** [review]

**Comment by nedaszilinskas** [comment]
@aisopuro I've added your comment/explanation from that issue to a readme.

As for myself, it was a little bit of a struggle at first to get a barcode to print (as it depends on so many conditions, needed to look into code directly to follow). Do you think we should also list these conditions in `l10n_fi_invoice` readme (barcode func. comes from another module - `l10n_fi_bank_barcode`)?

It is added here though https://github.com/avoinsystems/avoinsystems-addons/blob/13.0/l10n_fi_bank_barcode/__manifest__.py#L28. Do you think that's enough?

**Comment by aisopuro** [comment]
> @aisopuro I've added your comment/explanation from that issue to a readme.

Nice, thanks. Only concern now is does it look readable to users: since it isn't in `rst`, I think it might be a little hard to parse in Odoo's UI when looking at the module there. So if that readability is important, we might want to transform it to `rst`.

> As for myself, it was a little bit of a struggle at first to get a barcode to print (as it depends on so many conditions, needed to look into code directly to follow). Do you think we should also list these conditions in `l10n_fi_invoice` readme (barcode func. comes from another module - `l10n_fi_bank_barcode`)?
> 
> It is added here though https://github.com/avoinsystems/avoinsystems-addons/blob/13.0/l10n_fi_bank_barcode/__manifest__.py#L28. Do you think that's enough?

That seems fine, if it's visible in Odoo's UI. You might consider referrring to it from `l10n_fi_invoice`'s README. I think we can assume that people wanting to use that PDF will often want barcodes, so adding a line like 
> For instructions on how to get bank barcodes to be generated on the print, see the `l10n_fi_bank_barcode` module.

should be fine.

**Comment by nedaszilinskas** [comment]
> 

I guess in this case we need to add it to `index.html` as of now it is what's shown on the UI (not README.md nor README.rst).

**Comment by nedaszilinskas** [comment]
> > 
> 
> I guess in this case we need to add it to `index.html` as of now it is what's shown on the UI (not README.md nor README.rst).

I have updated the index.html
